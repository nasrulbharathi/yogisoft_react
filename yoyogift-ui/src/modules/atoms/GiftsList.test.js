import React from "react";
import { shallow } from "enzyme";
import { GiftsList } from "./GiftsList";

describe("Gifts List ", () => {
  const giftCardsFilteredValue = [
    {
      cardName: "testcard",
      cardPoints: 123,
      receiverEmail: "test@test.com",
      cardIssueDate: "Mon May 04 2020 14:51:28 GMT+0530 (India Standard Time)",
      cardExpiryDate: "Mon Aug 04 2020 14:51:28 GMT+0530 (India Standard Time)",
    },
  ];
  let p = {
    classes: {},
    giftCardsFiltered: giftCardsFilteredValue,
    userDetails: {
      isAdmin: true,
    },
  };
  it("should be rendered", () => {
    const wrapper = shallow(<GiftsList {...p} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should Invoke handleChangePage", () => {
    const wrapper = shallow(<GiftsList {...p} />);
    const instance = wrapper.instance();
    const event = {
      target: {
        value: 5,
      },
    };
    instance.handleChangePage(event, 5);
    expect(instance.state.page).toBe(5);
  });
  it("should Invoke handleChangeRowsPerPage", () => {
    const wrapper = shallow(<GiftsList {...p} />);
    const instance = wrapper.instance();
    const event = {
      target: {
        value: 5,
      },
    };
    instance.handleChangeRowsPerPage(event);
    expect(instance.state.rowsPerPage).toBe(5);
  });
});
