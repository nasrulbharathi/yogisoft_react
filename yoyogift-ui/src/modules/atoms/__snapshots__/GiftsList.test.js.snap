// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gifts List  should be rendered 1`] = `ShallowWrapper {}`;
